### PR TITLE
Null Checking for creating new ideas

### PR DESCRIPTION
**Pull request by Mctalian**
This handles null checking for new ideas. There are still a lot of other endpoints that need null-checking.

@YashdalfTheGray @alanlgirard @bcbrennecke @monotkate @davethenipper 

- [x] Code 1 @bcbrennecke
- [ ] Code 2 ([INSERT_NAME_HERE])
- [ ] Func 1 ([INSERT_NAME_HERE])
- [ ] Func 2 ([INSERT_NAME_HERE])

### CODE REVIEW

**Comment by bcbrennecke** [comment]
Code looks good.

**Comment by alanlgirard** [comment]
Postman says your code is functioning properly!  Also code looks good.

Approved! :+1:

File: server/ideas/ideaModel.js [hunk 1]
```diff
@@ -5,6 +5,25 @@ function Idea(title, description, authorId, eventId, tags, rolesreq) {
...
+    if (!description) {
+        throw new TypeError("Invalid description");
+    }
+    if (!authorId) {
+        throw new TypeError("Invalid authorId");
+    }
+    if (eventId === null || typeof eventId === 'undefined') {
+        throw new TypeError("Invalid eventId");
+    }
+    if (!tags) {
```
Comment (alanlgirard) [inline server/ideas/ideaModel.js]: This protects if null is passed in for the tags in general but doesn't protect against the actual tags in the array being null.

Protects
`"tags":null`

Not Protected
`"tags":[null]`

File: server/ideas/ideaModel.js [hunk 2]
```diff
@@ -5,6 +5,25 @@ function Idea(title, description, authorId, eventId, tags, rolesreq) {
...
+        throw new TypeError("Invalid tags");
+    }
+    if (!rolesreq) {
```
Comment (alanlgirard) [inline server/ideas/ideaModel.js]: Same goes for rolesreq.  The inner items are not protected from null.

Protected
'"rolesreq":null'

Not protected
'"rolesreq":[{"name":null,"_lowername":null,"checked":null}]'

File: server/ideas/ideaModel.js [hunk 3]
```diff
@@ -5,6 +5,25 @@ function Idea(title, description, authorId, eventId, tags, rolesreq) {
 
     var now = new Date().toISOString();
 
+    if (!title) {
+        throw new TypeError("Invalid title");
+    }
+    if (!description) {
+        throw new TypeError("Invalid description");
+    }
+    if (!authorId) {
```
Comment (alanlgirard) [inline server/ideas/ideaModel.js]: This also doesn't seem to be protecting against an authorId of null for some reason

**Comment by Mctalian** [comment]
This is a broader issue. I'm closing this PR until we can discuss and assess the proper way to protect against all nulls.

**Pull request closed without merging**